src/Bearded.TD.Generators/Events/EventHandlerGenerator.cs

```diff
@@ -0,0 +1,187 @@
+using System;
+using System.Collections.Generic;
+using Bearded.TD.Generators.Types;
+using Bearded.TD.Shared.Events;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Collections.Immutable;
+using System.Linq;
+using static Bearded.TD.Generators.SourceTemplates;
+
+namespace Bearded.TD.Generators.Events;
+
+[Generator]
+public class EventHandlerGenerator : IIncrementalGenerator
+{
+    private const string eventBaseTypeName =
+        "global::Bearded.TD.Game.Simulation.GameObjects.IComponentEvent";
+
+    private static readonly ImmutableArray<string> defaultNamespaces =
+        ImmutableArray.Create(
+            "Bearded.TD.Shared.Events"
+        );
+
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        var handlerMethods =
+            context.SyntaxProvider.ForAttributeWithMetadataName(
+                    typeof(HandlerAttribute).FullName!,
+                    predicate: static (node, _) => isValidHandlerMethod(node),
+                    transform: static (ctx, _) => handlerMethodFromContext(ctx))
+                .WhereNotNull()
+                .Collect()
+                .SelectMany(static (methods, _) => groupByClass(methods))
+                .WithComparer(new OrderIndependentClassWithHandlersComparer());
```

Comment:
Why do we use an order-independent comparer if we sort the methods when grouping them?